src/main.rs

```diff
@@ -51,25 +55,26 @@ async fn main() -> anyhow::Result<()> {
         command,
     } = Args::parse();
 
+    // Expand vector as more CRDs are created.
+    let crds = vec![kubit::resources::AppInstance::crd()];
     match &command {
-        Some(Commands::Manifests { crd_dir }) => match crd_dir {
-            Some(crd_dir) => {
-                // Expand vector as more CRDs are created.
-                for crd in vec![&kubit::resources::AppInstance::crd()] {
-                    let crd_file = format!("{}_{}.yaml", crd.spec.group, crd.spec.names.plural);
-                    let crd_path = PathBuf::from(crd_dir).join(crd_file);
-
-                    let file =
-                        File::create(crd_path).expect("Could not open AppInstances CRD file");
-
-                    serde_yaml::to_writer(&file, &kubit::resources::AppInstance::crd())?;
-                }
+        Some(Commands::Manifests { crd_dir }) => {
+            for crd in crds {
+                let mut out_writer = match crd_dir {
+                    Some(dir) => {
+                        let crd_file = format!("{}_{}.yaml", crd.spec.group, crd.spec.names.plural);
+                        let crd_path = PathBuf::from(dir).join(crd_file);
+                        let file =
+                            File::create(crd_path).expect("Could not open AppInstances CRD file");
+                        Box::new(file) as Box<dyn Write>
+                    }
+                    None => Box::new(stdout()) as Box<dyn Write>,
+                };
+                // The YAML delimiter is added in the event we have multiple documents.
+                out_writer.write(b"---\n").unwrap();
```

Comment:
This seemed simpler than running a `serde_yaml::to_string` and then using `.push_str("---\n")` or so, i.e. dealing with the different types of `&str` and `CustomResourceDefinition`. It was also giving me weird results from using the `serde_yaml::to_string` into a `to_writer`.

Happy to alter this though if there is more elegant way to handle it.

Reply:
1. `dir` is already a PathBuf, no need to `.from(dir)`
2. you can avoid specifying `as Box<dyn Write>` twice in the conditional branches, by just annotating the type of `let mut out_writer: Box<dyn Write>`

----

I found that an IDE that shows the inferred type of expressions (most do, I've tried clion, vscode and vim+rustanalyzer) is very useful:

<img width="206" alt="image" src="https://github.com/kubecfg/kubit/assets/52673/f11cd515-87dc-4d6d-92ad-199a43745bc0">

I'll just add a commit to your branch and merge